packages/engine-server/src/drivers/sqlite/tables/HierarchyTableUtils.ts

```diff
@@ -133,4 +133,33 @@ export class HierarchyTableUtils {
       }
     });
   }
+
+  /**
+   * Given a domain,
+   * recursively grab all descendants of the domain
+   * until we run out
+   *
+   * return all (parent, child) pairs
+   *
+   * more on recursive CTEs: https://www.sqlite.org/lang_with.html
+   */
+  public static getAllInDomain(
```

Comment:
How about `getAllDescendants()` as a name? Since we're just dealing with a concept of ID's here, I don't think we need to user the word 'domain'.